core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -990,6 +1041,22 @@ private void processBind(@Transient ObjList<BindVariableSetter> bindVariableSett
         hi = getStringLength(lo, msgLimit);
         checkNotTrue(hi == -1, "bad prepared statement name length [msgType='B']");
 
+        CharSequence statementName = getStatementName(lo, hi);
+        if (statementName != null) {
+            final Object statement = portalFactoryCache.peek(statementName);
```

Comment:
it is possible that `peek` will yield null with associative cache. Remember that cache liable to loose values when saturated.

There are two options here:

- we chose different data structure to hold cache information about named statements and clear this structure when named statements go away somehow. Perhaps client closes them?

- less graceful, but if we can report something on the wire about not being able to find statement by given name - we should do that